lib/src/Pos/Diffusion/Full.hs

```diff
@@ -83,27 +67,39 @@ instance (Given (m BlockVersionData)) => HasAdoptedBlockVersionData m where
 -- That's to say, we'd have to do the same work anyway, but then even more
 -- work to juggle the instances.
 diffusionLayerFull
-    :: forall d x .
+    :: forall d m x .
        ( DiffusionWorkMode d
        , MonadFix d
+       , MonadIO m
+       , WithLogger m
+       , MonadMask m
+       , MonadCatch m
+       , MonadThrow m
+       , Mockable Fork m
```

Comment:
These constraints, except `WithLogger`, are redundant, because they are all part of `DiffusionWorkMode`.